src/widget.cpp

```diff
@@ -358,6 +358,37 @@ static inline void DrawImageButtons(const Rect &r, WidgetType type, Colours colo
 	DrawSpriteIgnorePadding(img, PAL_NONE, r, align);
 }
 
+/**
+ * Draw a button with image and rext.
+ * @param r       Rectangle of the button.
+ * @param colour  Colour of the button.
+ * @param clicked Button is clicked.
+ * @param img     Image caption.
+ * @param text_colour Colour of the text.
+ * @param text    Text caption.
+ * @param align   Alignment of the caption.
+ * @param fs      Font size of the text.
+ */
+static inline void DrawImageTextButtons(const Rect &r, Colours colour, bool clicked, SpriteID img, TextColour text_colour, const std::string &text, StringAlignment align, FontSize fs)
+{
+	DrawFrameRect(r.left, r.top, r.right, r.bottom, colour, (clicked) ? FrameFlag::Lowered : FrameFlags{});
+
+	bool rtl = _current_text_dir == TD_RTL;
+	int image_width = img != 0 ? GetScaledSpriteSize(img).width : 0;
+	Rect r_img = r.Shrink(WidgetDimensions::scaled.framerect).WithWidth(image_width, rtl);
+	Rect r_text = r.Shrink(WidgetDimensions::scaled.framerect).Indent(image_width + WidgetDimensions::scaled.hsep_wide, rtl);
+
+	if (img != 0) {
+		DrawSpriteIgnorePadding(img, PAL_NONE, r_img, SA_LEFT | (align & SA_VERT_MASK));
```

Comment:
Align the sprite with `SA_HOR_CENTER` instead of `SA_LEFT`. It will still be left aligned but horizontally centred within its slot. This looks better at non-integer scales, IMHO.

![image](https://github.com/user-attachments/assets/1b9ec093-3306-470d-bae1-dab76cfb2a08)
vs
![image](https://github.com/user-attachments/assets/81049e1d-cb9d-4c8c-b315-14f15ecd6523)